### PR TITLE
Desktop notification

### LINKED ISSUE #581: Desktop Daily Top News Notification App
# Aim

- a simple application which produces a notification message in the form of a pop-up message on desktop. It runs on your system and will be utilized to send you notifications after some time. notification on desktop will be top news headlines of the day which will be fetched from news website

# Details

- libraries like request, notify2, etc will be used to make this application

### PR DESCRIPTION
**Pull request by Amit366**
# Description

Desktop notification

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
@Amit366 You missed this line in the issue description: 
```
notification on desktop will be top news headlines of the day which will be fetched from news website
```

**Comment by Amit366** [comment]
@kaustubhgupta in a url how can I replace api key with api key file??

File: Desktop Notifier/script.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import time
+from plyer import notification
+import requests
+import json
+
+
+
+news = requests.get('https://newsapi.org/v2/top-headlines?country=in&apiKey=51606761f68a4b20b4296e61814dbd1a')
```
Comment (kaustubhgupta) [inline Desktop Notifier/script.py:8]: take input for api key, and country from the user before notifying

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Script doesn't work:
image


Also, rename this folder as Desktop News Notifier along with these changes:

File: Desktop Notifier/script.py (hunk 1 above)
Comment (Amit366) [reply Desktop Notifier/script.py:8]: @kaustubhgupta  sir how to write the variable name in the url?? 
Please help me with it.

Comment (kaustubhgupta) [reply Desktop Notifier/script.py:8]: as a URL is string itself, you can use the concept of f-strings. Example:
```python
video_id = "bqOfdv9i4sE"
URL = f"https://www.youtube.com/watch?v={video_id}"
```

**Comment by Amit366** [comment]
@kaustubhgupta once check now

**Review by kaustubhgupta — APPROVED** [review]
It's not working:

image


Did you run the script on your local machine before pushing the code here?

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
image

**Comment by Amit366** [comment]
@kaustubhgupta it's showing string too long. It's not an error

**Comment by Amit366** [comment]
How to overcome that??

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta it's showing string too long. It's not an error

It's "ValueError", not a warning. Didn't you test this on your system?

**Comment by Amit366** [comment]
I tested it. It was the same. Now what should I do to change it?? @kaustubhgupta

**Comment by kaustubhgupta** [comment]
> I tested it. It was the same. Now what should I do to change it?? @kaustubhgupta

As the maximum character limit is 64, maybe trimming the text would work

**Comment by Amit366** [comment]
@kaustubhgupta trimming is just to remove whitespaces but I need to bring it to 64 bits. is there any way??

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta trimming is just to remove whitespaces but I need to bring it to 64 bits. is there any way??

Sorry wrong word, here I mean shorten/slice the news to 64 characters

**Comment by Amit366** [comment]
@kaustubhgupta its showing error as cannot be applied to int

**Comment by kaustubhgupta** [comment]
Why your title and message is int? If you convert them to str, then simply slice them string[:n], where n is the max value you want to supply in order to stay in limits of plyer

**Comment by Amit366** [comment]
I tried to convert into str but it's  the same

**Comment by Amit366** [comment]
@kaustubhgupta what to do now??

File: Desktop News Notifier/script.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+api_key = input("Enter the api key: ")
+
+news = requests.get(f'https://newsapi.org/v2/top-headlines?country={country_code}&apiKey={api_key}')
+
+data = json.loads(news.content)
+
+for i in range(10):
+
+    notification.notify(
+       title = data['articles'][i]['title'],
```
Comment (kaustubhgupta) [inline Desktop News Notifier/script.py:16]:
```suggestion
       title = data['articles'][i]['title'][:20],
```

File: Desktop News Notifier/script.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+       message= data['articles'][i]['description'] ,
```
Comment (kaustubhgupta) [inline Desktop News Notifier/script.py:17]:
```suggestion
       message= data['articles'][i]['description'][:44] ,
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Increase Timeout to 5 seconds and remove the time.sleep statement at the end. Also, as the time module is no longer needed, remove it's import statement

**Comment by Amit366** [comment]
@kaustubhgupta updated it

**Review by kaustubhgupta — APPROVED** [review]

**Comment by kaustubhgupta** [comment]
@antrikshmisri @santushtisharma10 @Kushal997-das

**Review by antrikshmisri — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM

**Pull request merged**